app/src/features/apiClient/screens/apiClient/components/clientView/components/request/components/RequestTabs/RequestTabs.tsx

```diff
@@ -58,18 +58,23 @@ const RequestTabs: React.FC<Props> = ({
     const isRequestBodySupported = supportsRequestBody(requestEntry.request.method);
     const isScriptsSupported = isApiClientScripts;
 
+    const autoGeneratedQueryParams = processAuthOptions(requestEntry.auth).queryParams;
+    const autoGeneratedHeaders = processAuthOptions(requestEntry.auth).headers;
+    const headersCount = sanitizeKeyValuePairs(requestEntry.request.headers).length + autoGeneratedHeaders.length;
+    const queryParamsCount =
+      sanitizeKeyValuePairs(requestEntry.request.queryParams).length + autoGeneratedQueryParams.length;
+
     const items = [
       {
         key: Tab.QUERY_PARAMS,
-        label: (
-          <LabelWithCount label="Query Params" count={sanitizeKeyValuePairs(requestEntry.request.queryParams).length} />
-        ),
+        label: <LabelWithCount label="Query Params" count={queryParamsCount} />,
         children: (
-          <KeyValueTable
+          <KeyValueTableWIthAutoGeneratedValues
             data={requestEntry.request.queryParams}
             setKeyValuePairs={setRequestEntry}
             pairType={KeyValueFormType.QUERY_PARAMS}
             variables={variables}
+            autoGeneratedValues={autoGeneratedQueryParams}
```

Comment:
We should create a `RequestQueryParamsTab` component which is responsible for the following things
- Fetching processed auth params
- Rendering components
    - <KeyValuePair  data={processedAuthParams} editable={false} />
    - <KeyValuePair  data={entry.params} editable={true} />

Same for headers too. This will keep the logic in the specific components only for rendering the tab

Reply:
For rendering the label count, we can expose a handler which updates the count in the parent component

Reply:
We can create different components for rendering the components but for label count it should stay in requestTabs component because count is shown on first mount for each tab if that logic is put in specific component then it will only change count when component tab is clicked. If you have any other approach please let me know.